### ISSUE TITLE
Build for 2.1.0 failing

### ISSUE THREAD

**Issue by smolinari** [orientechnologies/orientdb#4840]
Following the installation instructions in the manual. I have tried both maven and ant and both fail. 

Here is what ant outputs.

BUILD FAILED
/tmp/orientdb/build.xml:133: The following error occurred while executing this line:
/tmp/orientdb/build.xml:21: The following error occurred while executing this line:
/tmp/orientdb/core/build.xml:38: Compile failed; see the compiler error output for details.

See attached log.jpg file for complete output of compilation. Download it and rename it to log.txt. Github doesn't support .txt files. 

Scott

![log](https://cloud.githubusercontent.com/assets/3314957/9427849/5f45de2e-4991-11e5-8dce-07f07f1cbb9a.jpg)

**Comment by smolinari** [comment]
Oh, and I tried maven, because the ant install says it will be deprecated in 3.0 and to use maven.

Scott

**Comment by smolinari** [comment]
I also noticed the Travis builds are also breaking. Not sure if my problem has to do with them, but it would be nice to know, if the problem is in my court or yours. 

Here are some of the PRs with broken Travis builds. 

https://github.com/orientechnologies/orientdb/pull/4841
https://github.com/orientechnologies/orientdb/pull/3976
https://github.com/orientechnologies/orientdb/pull/4552

Scott

**Comment by nagarajasr** [comment]
which branch/tag are you trying to build? i was able to make a successful build without any issues on 2.1.0 GA. seems to me an environmental issue than anything else.

**Comment by smolinari** [comment]
I tried with 2.1.0. 

How can I check my environment? I have Debian Wheezy and this Java.

java version "1.6.0_36"
OpenJDK Runtime Environment (IcedTea6 1.13.8) (6b36-1.13.8-1~deb7u1)
OpenJDK 64-Bit Server VM (build 23.25-b01, mixed mode)
 
Scott

**Comment by lvca** [comment]
@smolinari Could you use Oracle JDK 8?

**Comment by smolinari** [comment]
Ok. Got Oracle Java 8 running and the build finished successfully and It looks like I can now start up OrientDB properly and all is working. (Might need a requirement note in the docs about minimum Java needed).

One thing. I can't access Studio. Is it correct that the ports ODB is on only bind to IPv6 (TCP6)?

Scott

**Comment by lvca** [comment]
Added in readme.txt.

**Comment by lvca** [comment]
The port should be IPv4

**Comment by smolinari** [comment]
Ok. As it looks, the Oracle Java 8 repo I installed only binds to IPv6 ports. I did some googling and found out I would need to set a Java environment variable. 

`_JAVA_OPTIONS="-Djava.net.preferIPv4Stack=true"`

So I entered that into my `/etc/environment` file and restarted the server. Although I could see Orient was now on IPv4 tcp ports 2480 and 2424, I still couldn't access Studio. I then realized, I probably need to open the ports in the firewall, so I made those changes and now all is working as should be. 

Scott
